### ISSUE TITLE
StoredProject.java:33-35: Implement the deactivate method...

### ISSUE THREAD

**Issue by 0pdd** [self-xdsd/self-core#42]
The puzzle `31-902d1319` from #31 has to be resolved: 

https://github.com/self-xdsd/self-core/blob/57732c637f0b9f8983317232da525f1ff7c716d9/self-api/src/main/java/com/selfxdsd/api/StoredProject.java#L33-L35

The puzzle was created by @amihaiemil on 19-May-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0pdd** [comment]
@amihaiemil please pay attention to this new issue.

**Comment by criske** [comment]
@amihaiemil I think  we have all the pieces to implement this.
We'll need these methods most likely:
 - `Tasks#removeAll()`  we need to deregister all project tasks first.
- `Contracts#removeAll()` remove all contracts associated with the project.
- `Projects#deregister(fullName,provider)` - the actual transaction that will remove all project tasks, contract tasks and delete the project from storage. 

Any thoughts?

**Comment by amihaiemil** [comment]
@criske Well, yes, but project deactivation will be a bit more complex. If we have active (unpaid) Invoices and a real Wallet set up, we should not allow project removal until the invoices are paid.

We should also not remove or unassign any ongoing Tasks, I think.

I think we should have a "Pause" button, to stop the PM from registering/assigning any new tasks. Then, the owner will wait until all ongoing tasks are invoiced and paid and only after that should we allow project deregistration.

Makes sense?

**Comment by criske** [comment]
@amihaiemil Makes sense, thanks :)
